### PR TITLE
More battle fixes

### PR DESCRIPTION
**Pull request by Ghabry**
To explain the not so obvious Things:
Due to changes to make the z-value working correctly only the last anim Frame was rendered. The BattleAnimation inherits from Sprite now which allows direct drawing on the Screen in that for-loop.
Seems to work fine only had to rename some conflicting functions.

The RNG Change:
I had a skill with 50% hit rate that basicly almost missed. I read on stackoverflow that rand with modulo can result in a bad Distribution. Therefore I switched this to the modern C++11 Api.

### CODE REVIEW

File: src/utils.cpp [hunk 1]
```diff
@@ -284,3 +289,12 @@ bool Utils::IsBigEndian() {
 
     return(d.c[0] == 1);
 }
+
+int32_t Utils::GetRandomNumber(int32_t from, int32_t to) {
+	std::uniform_int_distribution<int32_t> dist(from, to);
```
Comment (Zegeri) [inline src/utils.cpp]: Using int32_t as the numeric type might be undefined behavior. From the [documentation](http://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution):
>The effect is undefined if this is not one of short, int, long, long long, unsigned short, unsigned int, unsigned long, or unsigned long long. 

See also [this](https://groups.google.com/a/isocpp.org/forum/#!msg/std-discussion/rOzm69RSNcg/aIqYf__XJb0J)

**Comment by Zegeri** [comment]
By the way, could you change [this line](https://github.com/Ghabry/easyrpg-player/blob/battle/src/game_battlealgorithm.cpp#L758)? It should be:
```
this->agility = effect;
```

**Comment by fdelapena** [comment]
jenkins: retest this please :heart:

**Comment by fdelapena** [comment]
jenkins: retest this please thx

**Comment by fdelapena** [comment]
Jenkins: retest this please

**Comment by Ghabry** [comment]
Jenkins: test this please

**Comment by Ghabry** [comment]
Jenkins: test this please

**Comment by carstene1ns** [comment]
EV0002: hook this please... :grinning:

**Comment by Ghabry** [comment]
Jenkins: Test this please

**Comment by carstene1ns** [comment]
jenkins: test this please x3

**Comment by Ghabry** [comment]
Jenkins: Test this please

**Comment by Ghabry** [comment]
Jenkins: Test this please

**Comment by fdelapena** [comment]
jenkins: retest this please

**Comment by fdelapena** [comment]
jenkins: retest this please

**Comment by fdelapena** [comment]
Jenkins: retest this please

**Comment by Ghabry** [comment]
Ara Fell: Attacking all enemies with a multi target skill and healing all actors.

animation

UiD: Start animation and multi target animation

animation2

**Comment by Ghabry** [comment]
This needs a retest of that Hentai Film game #906 because of the Y-BattleAnimation offset change.
At least Ara Fell needs the "/ 3" but maybe it depends on another editor setting if it's / 3 or / 2.

**Comment by carstene1ns** [comment]
Note: You changed `BattleAnimationBattlers::Draw()`, while Chris changed `BattleAnimationChara::Draw()`.

File: src/battle_animation.cpp [hunk 2]
```diff
@@ -103,22 +95,10 @@ bool BattleAnimation::IsDone() const {
...
 
-Sprite* BattleAnimation::GetSprite() {
-	return sprite.get();
-}
-
 void BattleAnimation::OnBattleSpriteReady(FileRequestResult* result) {
 	if (result->success) {
-		sprite.reset(new Sprite());
-
-		//Normally only battle2 sprites are "large" sprites - but the check doesn't hurt.
```
Comment (ChristianBreitwieser) [inline src/battle_animation.cpp]: The changes from the fix for #906 are undone in this file.

**Comment by carstene1ns** [comment]
Yep, needs a rebase.

File: src/battle_animation.cpp [hunk 3]
```diff
@@ -250,18 +233,13 @@ BattleAnimationChara::~BattleAnimationChara() {
 	Graphics::RemoveDrawable(this);
 }
 void BattleAnimationChara::Draw() {
-	//If animation is targeted on the screen
-	if (animation.scope == RPG::Animation::Scope_screen) {
-		DrawAt(SCREEN_TARGET_WIDTH / 2, SCREEN_TARGET_HEIGHT / 2);
-		return;
```
Comment (ChristianBreitwieser) [inline src/battle_animation.cpp]: Sorry, but the code which fixes an animation targeting the screen on a map is still missing ;)

Comment (Ghabry) [inline src/battle_animation.cpp]: This is what happens when you use a diff tool you are not familiar with while rebasing >.<

**Pull request merged**